src/settings/LanguageLocalization/LanguageLocalization.js

```diff
@@ -39,50 +43,75 @@ const LanguageLocalization = () => {
     key: tenantLocaleConfig.KEY,
   });
 
+  const {
+    settings: userSettings,
+    updateSetting: updateUserSetting,
+  } = useSettings({
+    scope: userOwnLocaleConfig.SCOPE,
+    key: userOwnLocaleConfig.KEY,
+    userId: stripes.user?.user?.id,
+  });
+
   const ConnectedConfigManager = useMemo(() => stripes.connect(ConfigManager), [stripes]);
   const paneTitle = intl.formatMessage({ id: 'ui-myprofile.settings.languageLocalization.label' });
+  const userHasSetLocalePreferences = !!userSettings[fieldNames.LOCALE];
 
-  const localesOptions = useMemo(() => {
-    return localesList(intl, tenantSettings[fieldNames.LOCALE]);
-  }, [intl, tenantSettings]);
+  // tenant configuration might not have a set locale - try stripes-config locale and default to en-US
+  const tenantLocale = tenantSettings[fieldNames.LOCALE] || config.locale || 'en-US';
+  const localesOptions = useMemo(() => localesList(intl, tenantSettings[fieldNames.LOCALE]), [intl, tenantSettings]);
 
   const formatPayload = (newSettings) => {
     return {
-      ...initialSettings.current,
+      [fieldNames.LOCALE]: initialSettings.current,
       ...newSettings,
     };
   };
 
   const afterSave = ({ value: settings }) => {
-    const userLocale = settings[fieldNames.LOCALE];
-    const tenantLocale = tenantSettings[fieldNames.LOCALE];
-
-    let locale = settings[fieldNames.LOCALE];
-
-    let numberingSystem = settings.numberingSystem;
-
-    if (userLocale === tenantLocale) {
-      locale = tenantLocale;
-      numberingSystem = tenantSettings.numberingSystem;
-    }
-
+    const locale = settings[fieldNames.LOCALE];
+    // user can't select numbering system in My Profile, so we can just use tenant settings
```

Comment:
The user can select a numbering system in the Developer -> User locale, it will be present in user settings.